### PR TITLE
add missing dep on puppet/mosquitto

### LINKED ISSUE #751: 22.1.0 release is missing dep on puppet/mosquitto
It appears that [templates/plugin/remote_execution_ssh.yml.erb](https://github.com/theforeman/puppet-foreman_proxy/blob/4221f8a5e86a2aafb61eb9e1497f34e620451fa9/templates/plugin/remote_execution_ssh.yml.erb) is causing `foreman_proxy::plugin::remote_execution::mosquitto` to be evaluated regardless if it the class is declared in the manifest.

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Could not find declared class mosquitto (file: /etc/puppetlabs/code/environments/IT_3497_foreman3/modules/foreman_proxy/manifests/plugin/remote_execution/mosquitto.pp, line: 36, column: 3) on node foreman.dev.XXX
```

Which then fails as `puppet/mosquitto` isn't a declared dependency in `metadata.json`.

### PR DESCRIPTION
**Pull request by jhoblitt**
Resolves #751

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -45,6 +45,10 @@
     {
       "name": "richardc/datacat",
       "version_requirement": ">= 0.6.0 < 1.0.0"
+    },
+    {
+      "name": "puppet/mosquitto",
+      "version_requirement": ">= 1.0.0 < 2.0.0"
```
Comment (evgeni) [inline metadata.json:51]: Can you make this >= 1.1.0 please? It contains fixes we need.

**Review by evgeni — COMMENTED** [review]
Ah, good find! We wanted to keep that optional, but with the cleanup that Eric implemented it's not optional anymore.

File: metadata.json (hunk 1 above)
Comment (jhoblitt) [reply metadata.json]: done.

**Comment by jhoblitt** [comment]
It may be a good idea to change the beaker setup to `configure_beaker(modules: :metadata)` so metadata errors will be caught in the future.

**Review by evgeni — APPROVED** [review]

**Pull request merged**